### PR TITLE
Add Kafka BOM to Micronaut 3.5.0

### LINKED ISSUE #7116: Add kafka BOM to core
### Issue description

https://github.com/micronaut-projects/micronaut-kafka/issues/509

### PR DESCRIPTION
**Pull request by timyates**
[skip ci]

Blocked until https://github.com/micronaut-projects/micronaut-kafka/pull/510 is released as 4.3.0

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -255,9 +256,7 @@ managed-jna = { module = "net.java.dev.jna:jna", version.ref = "managed-jna" }
 
 managed-jsr305 = { module = "com.google.code.findbugs:jsr305", version.ref = "managed-jsr305" }
 
-managed-kafka = { module = "org.apache.kafka:kafka-clients", version.ref = "managed-kafka" }
 managed-kafka212 = { module = "org.apache.kafka:kafka_2.12", version.ref = "managed-kafka" }
```
Comment (sdelamo) [inline gradle/libs.versions.toml:259]: why do we have this? Do we know?

**Review by sdelamo — APPROVED** [review]

File: gradle/libs.versions.toml (hunk 1 above)
Comment (timyates) [reply gradle/libs.versions.toml:259]: I have no idea...  I originally deleted it, but I guess someone may be consuming it as a dependency, so I added it back in...

@graemerocher Do you recall where this is required?  I can find it added in 2018 here https://github.com/micronaut-projects/micronaut-core/commit/0f8c3f6682eb7d285af3acb7f8fd727ee116b7a7#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R80

My vote would be to delete it for 3.5.0, and see who squeaks

Comment (melix) [reply gradle/libs.versions.toml:259]: I don't think that `kafka212` is a great name either. Obviously encoding the version in the name of the alias kind of defeats the purpose. Maybe this should be reworked for 4.0.

**Review by sdelamo — APPROVED** [review]
I have released [Kafka 4.3.0](https://github.com/micronaut-projects/micronaut-kafka/releases/tag/v4.3.0). It should be good to go once CI passes. .

**Review by melix — APPROVED** [review]

**Comment by timyates** [comment]
@sdelamo I can't merge it, as the CLA isn't triggering 😢

**Pull request merged**